### PR TITLE
Upgrade omnipay-common to ^3.0

### PR DESCRIPTION
**Pull request by jsgv**

### CODE REVIEW

**Comment by abdala** [comment]
Hey! Thanks for the PR! :smiley:

File: src/Message/AbstractRequest.php [hunk 1]
```diff
@@ -73,6 +83,10 @@ protected function xml2array($xml)
     {
         $arr = [];
```
Comment (abdala) [inline src/Message/AbstractRequest.php]: Can we change this to a more meaningful name.
```suggestion
        $response = []; // or $result
```

File: src/Message/AbstractRequest.php [hunk 2]
```diff
@@ -73,6 +83,10 @@ protected function xml2array($xml)
...
 
+        if ($xml == false) {
```
Comment (abdala) [inline src/Message/AbstractRequest.php]: Maybe it is better to test for any invalid value here.

```suggestion
        if (! $xml) {
```

File: src/Message/TransactionSearchResponse.php [hunk 3]
```diff
@@ -12,9 +12,14 @@ public function __construct(RequestInterface $request, $data)
         parent::__construct($request, $data);
 
         $transactions = array();
+
         if ($this->isSuccessful()) {
-            foreach ($this->getData()['transactions']['transaction'] as $transaction) {
-                $transactions[] = $this->xml2array($transaction);
+            if ($this->getData()['resultsInThisPage'] == 1) {
```
Comment (abdala) [inline src/Message/TransactionSearchResponse.php:17]: I didn't understand why did you change this part of the code. Can you explain it a little bit, please?

Comment (jsgv) [reply src/Message/TransactionSearchResponse.php:17]: If `transactions` returned is only 1, it is a single element. Therefore a `foreach` fails.

File: src/Message/AbstractRequest.php (hunk 2 above)
Comment (jsgv) [reply src/Message/AbstractRequest.php]: Sounds good to me!

File: src/Message/AbstractRequest.php (hunk 1 above)
Comment (jsgv) [reply src/Message/AbstractRequest.php]: Sounds good to me!

**Review by abdala — APPROVED** [review]
